nes-nautilus/src/Nautilus/Interface/MemoryProvider/TupleBufferMemoryProvider.cpp

```diff
@@ -114,16 +115,23 @@ TupleBufferMemoryProvider::~TupleBufferMemoryProvider() = default;
 
 std::shared_ptr<TupleBufferMemoryProvider> TupleBufferMemoryProvider::create(const uint64_t bufferSize, const Schema& schema)
 {
-    if (schema.memoryLayoutType == Schema::MemoryLayoutType::ROW_LAYOUT)
-    {
-        auto rowMemoryLayout = std::make_shared<Memory::MemoryLayouts::RowLayout>(bufferSize, schema);
-        return std::make_shared<RowTupleBufferMemoryProvider>(std::move(rowMemoryLayout));
-    }
-    if (schema.memoryLayoutType == Schema::MemoryLayoutType::COLUMNAR_LAYOUT)
+    auto memoryLayout = Memory::MemoryLayouts::MemoryLayout::create(bufferSize, schema);
+
+    switch (schema.memoryLayoutType)
     {
-        auto columnMemoryLayout = std::make_shared<Memory::MemoryLayouts::ColumnLayout>(bufferSize, schema);
-        return std::make_shared<ColumnTupleBufferMemoryProvider>(std::move(columnMemoryLayout));
+        case Schema::MemoryLayoutType::ROW_LAYOUT: {
+            auto rowMemoryLayout
+                = NES::Util::as<Memory::MemoryLayouts::RowLayout>(Memory::MemoryLayouts::MemoryLayout::create(bufferSize, schema));
+            return std::make_shared<RowTupleBufferMemoryProvider>(RowTupleBufferMemoryProvider{rowMemoryLayout});
+        }
+        case Schema::MemoryLayoutType::COLUMNAR_LAYOUT: {
+            auto columnMemoryLayout
+                = NES::Util::as<Memory::MemoryLayouts::ColumnLayout>(Memory::MemoryLayouts::MemoryLayout::create(bufferSize, schema));
+            return std::make_shared<ColumnTupleBufferMemoryProvider>(ColumnTupleBufferMemoryProvider{columnMemoryLayout});
+        }
+            /// should never get accessed since creating the memoryLayout already checks schemas memoryLayoutType.
+        default:
+            throw NotImplemented("Currently only row and column layout are supported");
```

Comment:
I would not have these default cases. 

If you remove the default case, then anyone adding a new MemoryLayoutType is required to add an implementation. If you have a default case, this would compile, but throw at runtime.

Remove the default case and add a std::unreachable() after the switch to make it compile.